vllm/model_executor/layers/triton_kernel/prefix_prefill.py

```diff
@@ -618,7 +618,9 @@ def context_attention_fwd(q,
                               b_ctx_len,
                               max_input_len,
                               alibi_slopes=None):
-        BLOCK = 128
+
+        cap = torch.cuda.get_device_capability()
```

Comment:
Does prefix caching adapt other hardware? like AMD? This only considers cuda arch. Might it better that we define a global utility to get block size which handles different hardwares.

Reply:
I believe this kernel only works for NVIDIA right now. Let me merge this fix first and we can systematically test for AMD later.